util/types/json/functions.go

```diff
@@ -322,3 +322,44 @@ func set(j JSON, pathExpr PathExpression, value JSON, mt ModifyType) JSON {
 	// 3) we want to insert or replace something, but the path without last leg doesn't exist.
 	return j
 }
+
+// Remove removes the elements indicated by pathExprList from JSON.
+func (j JSON) Remove(pathExprList []PathExpression) (JSON, error) {
+	for _, pathExpr := range pathExprList {
+		if len(pathExpr.legs) == 0 {
+			// TODO: should return 3153(42000)
+			return j, errors.New("Invalid path expression")
+		}
+		if pathExpr.flags.containsAnyAsterisk() {
+			// TODO: should return 3149(42000)
+			return j, errors.New("Invalid path expression")
+		}
+		j = remove(j, pathExpr)
+	}
+	return j, nil
+}
+
+// remove is used in Remove.
+func remove(j JSON, pathExpr PathExpression) JSON {
+	currentLeg, subPathExpr := pathExpr.popOneLeg()
+	if currentLeg.typ == pathLegIndex && j.typeCode == typeCodeArray {
+		var index = currentLeg.arrayIndex
+		if len(j.array) > index {
```

Comment:
the index is parsed from string, and our pattern has already filtered negtive numbers.